Some extra options and fixes

Thulium-Drake opened the pull request:
* The manual mentioned VERSION and ARCH, but build uses B_VERSION and B_ARCH instead. Fixed that
* The torepo script (which is added for reference, I know :-) ) had a hardcoded repo server in it. I've moved that to toreporc
* Lastly, I've made a short guide on how to get started with build.

Commit:
Added extra options to B_GET git

This allows for stuff like checking out specific brances

Also, added a short installation guide and fixed some errors in BUILD.md

Commit:
Clarified wording in BUILD.md a bit more

Commit:
Added options for torepo to buildrc

* The $REPO variable in buildrc will now be used as the upload
destination for finished packages
* build will now also automatically any required directories as defined
in buildrc

Commit:
Fixed a derp, removed testing echo

mverboom commented:
Thanks for the PR.

With the extra options to B_GET for git, can you explain the difference to doing this and doing a git checkout branchname after the the repository has been cloned?

torepo is just an example of a script that could be run by build. It is not part of what build is trying to do. The cleanest solution would be to create something like a .toreporc if you want to make it more modular or improve functionality.

On the creation of directories:
CACHEDIR is automatically created when B_GET is used.
RECIPEDIR doesn't sound like a directory that should be automatically created. Build doesn't do anything without a recipe, so I would assume when you make a recipe you also put it in the folder you defined in the config file..
DESTDIR is automatically created during the run of a recipe.
TMPDIR is automatically created, I removed the check and error.
PKGDIR wasn't created, it will now be automatically created and there will be a warning.

Thanks,
Mark

Commit:
Updated stuff in regards of feedback on PR

* Removed creating directories
* Split torepo config to toreporc

Commit:
Merge branch 'master' of github.com:mverboom/build

Commit:
Reverted changes to B_GET

Commit:
Reverted changes to B_GET

Commit:
Merge branch 'master' of github.com:Thulium-Drake/build

Commit:
Merge branch 'master' of github.com:Thulium-Drake/build

Thulium-Drake commented:
Oh... Yeah, that makes sense, I didn't think of that. I was trying to figure out a way to make the checkout smarter, but the git command can do that itself :-) I've checked the man page of git clone, but as there isn't anything usefull

True, I've moved it to a different file. I was trying to avoid creating a lot of config files, but it makes sense to do so.

I've removed the directory creation step in regards of the fix you've made. I've also edited the first post to reflect the remaining changes

Commit:
Merge branch 'master' of github.com:Thulium-Drake/build

Commit:
Merge branch 'master' of github.com:Thulium-Drake/build

Commit:
Merge branch 'master' of github.com:Thulium-Drake/build

Commit:
Merge branch 'master' of github.com:Thulium-Drake/build

Commit:
Merge branch 'master' of git.element-networks.nl:Element-Networks/build

Commit:
Merge branch 'master' of git.element-networks.nl:Element-Networks/build

Commit:
Merge branch 'master' of git.element-networks.nl:Element-Networks/build

Commit:
Merge branch 'master' of git.element-networks.nl:Element-Networks/build

Commit:
Merge branch 'master' of git.element-networks.nl:Element-Networks/build

Commit:
Merge branch 'master' of git.element-networks.nl:Element-Networks/build

Commit:
Merge branch 'master' of git.element-networks.nl:Element-Networks/build

mverboom commented:
Merged!
